src/mm-form/mm-form.js

```diff
@@ -0,0 +1,490 @@
+			// config - values from config override domConfig
+			for (var key in config) {
+				var field = config[key].field || this._select('[name='+key+']'),
+					cfg = config;
+
+				if (!field) throw 'There must be a corresponding DOM element for config[\''+key+'\']';
+
+				cfg[key].errorMsgEleDOM = this._select('#'+cfg[key].errorMsgEle) || null;
+				cfg[key].parentEleDOM = this._select('#'+cfg[key].parentEle) || Polymer.dom(field).parentNode;
+				cfg[key] = StrandLib.DataUtils.copy(domConfig[key], cfg[key]);
+
+				// Create error message element
+				if (cfg[key].errorMsg && !cfg[key].errorMsgEle) {
+					this._createErrorMsg(key, cfg[key].errorMsg, cfg[key].parentEleDOM);
+				}
+
+				// Create the field label
+				if (cfg[key].label) {
+					this._createLabel(key, cfg[key].label, field, cfg[key].parentEleDOM);
+				}
+			}
+		},
+
+		_initData: function() {
+			if(this._isEmpty(this.config)) return;
+
+			for (var key in this.config) {
+				var field 	= this.config.field || this._select('[name='+key+']');
+
+				if (!field) {
+					throw 'There must be a corresponding DOM element for data[\''+key+'\']';
+				}
+
+				var exclude	= this.config[key].exclude || null,
+					value = this.data[key] || null;
+				
+				// If there was an initial value set in markup, use it
+				if (field.value && value === null) {
+					value = field.value;
+				}
+
+				// Update data
+				if (!exclude) this._updateData(key, value);
+				this._initialData[key] = value;
+
+				// Populate the fields if necessary
+				if (value && field.value !== value) {
+					field.value = value;
+				}
+			}
+		},
+
+		_createErrorMsg:function(key, errorMsg, parentEleDOM) {
+			var existingMsgEle = this._select('._'+key+'-error-msg') || null;
+
+			if (!existingMsgEle) {
+				// create one:
+				errorMsgEleDOM = new Strand.FormMessage();
+				errorMsgEleDOM.type = 'error';
+
+				errorMsgEleDOM.message = errorMsg;
+				errorMsgEleDOM.classList.add('_'+key+'-error-msg');
+				Polymer.dom(parentEleDOM).appendChild(errorMsgEleDOM);
+
+				// store the formMessage ref to config
+				this.config[key].errorMsgEleDOM = errorMsgEleDOM;		
+			} else {
+				existingMsgEle.message = errorMsg;
+			}
+		},
+
+		_createLabel:function(key, label, field, parentEleDOM) {
+			var existingLblEle 	= this._select('._'+key+'-label') || null,
+				formLabel 		= null, 
+				labelTxt 		= null;
+
+			if (!existingLblEle) {
+				// create one:
+				labelTxt = document.createTextNode(label);
+				formLabel = new Strand.Header();
+
+				formLabel.size = 'medium';
+				formLabel.setAttribute('name', name);
+				formLabel.classList.add('_'+key+'-label');
+				formLabel.setAttribute('form-header', true);
+
+				Polymer.dom(formLabel).appendChild(labelTxt);
+				Polymer.dom(parentEleDOM).insertBefore(formLabel, field);
+
+				// store the formLabel ref to config
+				this.config[key].formLabel = formLabel;
+			} else {
+				labelEle.innerHTML = null;
```

Comment:
Thanks @jcmoore - this was indeed incorrect, but exposed some underlying gotchas when switching config 'on-the-fly'. Updates forthcoming.